### PR TITLE
Simplify supervisor ports api

### PR DESCRIPTION
**Pull request by csweichel**
- [x] /werft ws-feature-flags=registry_facade
- [x] /werft https

This PR makes the supervisor port status API non-incremental.

We do this because
- all other status APIs within Gitpod are non-incremental, certainly within supervisor. This step makes the API more consistent.
- the previous incremental implementation could loose updates due to a race between the initially propagated status and subsequent updates.

### How to test
1. Start a workspace in Theia, and start a service on a previously unexposed port (e.g. `curl lama.sh | LAMA_PORT=9090 sh`)
2. Start a workspace with Code, and start a service on a previously unexposed port (e.g. `curl lama.sh | LAMA_PORT=9090 sh`)

### CODE REVIEW

File: components/theia/packages/gitpod-extension/src/browser/ports/gitpod-ports-service.ts [hunk 1]
```diff
@@ -53,26 +53,22 @@ export class GitpodPortsService {
         return this._ports.values();
     }
 
-    private updatePorts({ added, updated, removed, initial }: DidChangeGitpodPortsEvent): void {
+    private updatePorts({ ports, initial }: DidChangeGitpodPortsEvent): void {
         const toClean = initial ? new Set<number>(this._ports.keys()) : undefined;
```
Comment (akosyakov) [inline components/theia/packages/gitpod-extension/src/browser/ports/gitpod-ports-service.ts:57]: it should happen always now, we can drop `initial`

File: components/theia/packages/gitpod-extension/src/node/gitpod-port-server-impl.ts [hunk 2]
```diff
@@ -46,25 +46,14 @@ export class GitpodPortServerImpl implements GitpodPortServer {
                             this.ports.clear();
                             initial = false;
                         }
-                        let added: PortsStatus.AsObject[] | undefined
```
Comment (akosyakov) [inline components/theia/packages/gitpod-extension/src/node/gitpod-port-server-impl.ts:49]: `this.ports` can be just an array now which we always assign here

File: components/theia/packages/gitpod-extension/src/browser/ports/gitpod-ports-service.ts (hunk 1 above)
Comment (akosyakov) [reply components/theia/packages/gitpod-extension/src/browser/ports/gitpod-ports-service.ts:57]: the same in Code: https://github.com/gitpod-io/vscode/commit/875df34b9ecb5059346a2fd46d055fceb7803945#diff-0d515e0b70f335552688d0b80465bc1f0e9525cc759ec8f66371698612447767R163

**Comment by akosyakov** [comment]
Tasks API is also incremental, not sure whether we want to align it.

**Comment by csweichel** [comment]
> Tasks API is also incremental, not sure whether we want to align it.

Good point - sorted in #2276

**Review by akosyakov — APPROVED** [review]
LGTM, thank you!

**Pull request merged**